orangecontrib/text/preprocess.py

```diff
@@ -1,94 +1,298 @@
+            self.use_df_sw = True
+            is_float = isinstance(max_df, float)
+            is_int = isinstance(max_df, int)
+            if is_float:
+                in_range = 1.0 >= max_df >= 0.0
+                if not in_range:
+                    raise ValueError('Parameter max_df is out of range ({}).'
+                                     .format(max_df))
+            elif not is_int:
+                raise ValueError('Type "{}" not supported.'.format(max_df))
 
 
 class Stemmatizer():
-    """
-        A common class for stemming and lemmatization.
-    """
-    def __init__(self, trans, lowercase=True, name='Stemmatizer'):
+    """A common class for stemming and lemmatization."""
+    def __init__(self, transformation, name='Stemmatizer'):
         """
-            :param trans: The method that will perform transformation on the tokens.
-            :param name: The name of the transformation object.
-            :return: :class: `orangecontrib.text.preprocess.Stemmatizer`
+        Args:
+            transformation (function): The function that will perform
+                transformation on the tokens.
+            name (Optional[str]): The name of the transformation object.
+
+        Returns:
+            `orangecontrib.text.preprocess.Stemmatizer`: The instance of a
+                Stemmatizer of choice.
         """
-        self.trans = trans
+        self.transformation = transformation
         self.name = name
-        self.lowercase = lowercase
 
     def __call__(self, data):
-        """
-            :param data: The input that we wish to transform.
-            :type data: string, list or :class: `Orange.data.Table`
-            :return: string, list or :class: `Orange.data.Table`
+        """Calls the transformation function of this instance.
+
+        Args:
+            data (str, list): The input that we wish to transform.
+
+        Returns:
+            str, list: The transformed input in the form of a string or list,
+                depending on the input.
+
+        Raises:
+            ValueError: If the input type is not supported.
         """
         if isinstance(data, str):
-            if self.lowercase:
-                return self.trans(data).lower()
-            else:
-                return self.trans(data)
+            return self.transformation(str)
         elif isinstance(data, list):
-            if self.lowercase:
-                return [self.trans(word).lower() for word in data]
-            else:
-                return [self.trans(word) for word in data]
+            return [self.transformation(word) for word in data]
         else:
-            raise ValueError("Type {} not supported.".format(type(data)))
+            raise ValueError('Type {} not supported.'.format(type(data)))
 
-Stemmer = Stemmatizer(PorterStemmer().stem, 'Stemmer')
+PorterStemmer = Stemmatizer(PorterStemmer().stem, 'PorterStemmer')
+SnowballStemmer = Stemmatizer(SnowballStemmer(language="english").stem,
```

Comment:
For the `Preprocessor` class itself we wouldn't even need to reformat anything. I believe the parameter `stop_words` in a format (`str`, `list` or `None`) is sufficient. If it is `str` just pass it over to NLTK.

For the `Stemmatizer` however, we should change something. Maybe we could let the user create the Stemmatizers?

Anyhow, I suggest that you look at those minor changes first, so we can merge this ASAP. The support for other languages can also be added later in another PR.